### ISSUE TITLE
Remembered users list not returned in UWP simulator

### ISSUE THREAD

**Issue by Arslan007** [AzureAD/azure-activedirectory-library-for-dotnet#1586]
**Which Version of ADAL are you using ?**
ADAL 4.5.1

**Which platform has the issue?**
UWP, Xamarin

**What authentication flow has the issue?**
* Desktop / Mobile
    * [x] Interactive
    * [ ] Integrated Windows Auth
    * [ ] Username Password
    * [ ] Device code flow (browserless)
* Web App
    * [ ] Authorization code
    * [ ] OBO
* Web API
    * [ ] OBO

Other? - please describe;

**Is this a new or existing app?**
 This is a new app 

**Repro**
#1580 Fixes the issue when it runs on actual device but not working in Simulator

**Expected behavior**
When run in simulator for UWP then it should list of remembered user for that particular app who logged in earlier on same device and app

**Actual behavior**
When run in simulator for UWP then its not showing the list of remembered users. 

**Additional context/ Logs / Screenshots**

```
Information 2019-04-18T22:05:36.1756681Z: 1d70e22e-efgh-458e-abcd-12345d34d34a - AdalLoggerBase.cs: ADAL PCL.UAP with assembly version '5.0.1.0', file version '5.0.1.0' and informational version '5.0.1-internal10701ssofix' is running...
Information 2019-04-18T22:05:36.1851751Z: 1d70e22e-efgh-458e-abcd-12345d34d34a - AdalLoggerBase.cs: === Token Acquisition started:
	Authority: https://login.microsoftonline.com/Tenant/
	Resource: https://XYZ.com/Technician-WS-DEV
	ClientId: 299a1a70-xxxx-4c56-xxx-85a344xxx955c
	CacheType: Microsoft.IdentityModel.Clients.ActiveDirectory.TokenCache (0 items)
	Authentication Target: User
	
Verbose 2019-04-18T22:05:37.8851731Z: 1d70e22e-efgh-458e-abcd-12345d34d34a - AdalLoggerBase.cs: Loading from cache.
Verbose 2019-04-18T22:05:37.8996684Z: 1d70e22e-efgh-458e-abcd-12345d34d34a - AdalLoggerBase.cs: Looking up cache for a token...
Information 2019-04-18T22:05:37.9121711Z: 1d70e22e-efgh-458e-abcd-12345d34d34a - AdalLoggerBase.cs: No matching token was found in the cache
Information 2019-04-18T22:05:37.9141686Z: 1d70e22e-efgh-458e-abcd-12345d34d34a - AdalLoggerBase.cs: Checking MSAL cache for user token cache
Information 2019-04-18T22:05:37.9326681Z: 1d70e22e-efgh-458e-abcd-12345d34d34a - AdalLoggerBase.cs: A match was found in the MSAL cache ? False
Verbose 2019-04-18T22:05:37.9351758Z: 1d70e22e-efgh-458e-abcd-12345d34d34a - AdalLoggerBase.cs: Looking up cache for a token...
Information 2019-04-18T22:05:37.9371690Z: 1d70e22e-efgh-458e-abcd-12345d34d34a - AdalLoggerBase.cs: No matching token was found in the cache
Information 2019-04-18T22:05:37.9391694Z: 1d70e22e-efgh-458e-abcd-12345d34d34a - AdalLoggerBase.cs: Checking MSAL cache for user token cache
Information 2019-04-18T22:05:37.9416711Z: 1d70e22e-efgh-458e-abcd-12345d34d34a - AdalLoggerBase.cs: A match was found in the MSAL cache ? False
Verbose 2019-04-18T22:05:37.9441710Z: 1d70e22e-efgh-458e-abcd-12345d34d34a - AdalLoggerBase.cs: Looking up cache for a token...
Information 2019-04-18T22:05:37.9461705Z: 1d70e22e-efgh-458e-abcd-12345d34d34a - AdalLoggerBase.cs: No matching token was found in the cache
Information 2019-04-18T22:05:37.9486682Z: 1d70e22e-efgh-458e-abcd-12345d34d34a - AdalLoggerBase.cs: Checking MSAL cache for user token cache
Information 2019-04-18T22:05:37.9511846Z: 1d70e22e-efgh-458e-abcd-12345d34d34a - AdalLoggerBase.cs: A match was found in the MSAL cache ? False
Verbose 2019-04-18T22:05:37.9531660Z: 1d70e22e-efgh-458e-abcd-12345d34d34a - AdalLoggerBase.cs: Looking up cache for a token...
Information 2019-04-18T22:05:37.9556697Z: 1d70e22e-efgh-458e-abcd-12345d34d34a - AdalLoggerBase.cs: No matching token was found in the cache
Information 2019-04-18T22:05:37.9581718Z: 1d70e22e-efgh-458e-abcd-12345d34d34a - AdalLoggerBase.cs: Checking MSAL cache for user token cache
Information 2019-04-18T22:05:37.9611681Z: 1d70e22e-efgh-458e-abcd-12345d34d34a - AdalLoggerBase.cs: A match was found in the MSAL cache ? False
Verbose 2019-04-18T22:05:37.9631703Z: 1d70e22e-efgh-458e-abcd-12345d34d34a - AdalLoggerBase.cs: Looking up cache for a token...
Information 2019-04-18T22:05:37.9656718Z: 1d70e22e-efgh-458e-abcd-12345d34d34a - AdalLoggerBase.cs: No matching token was found in the cache
Information 2019-04-18T22:05:37.9681714Z: 1d70e22e-efgh-458e-abcd-12345d34d34a - AdalLoggerBase.cs: Checking MSAL cache for user token cache
Information 2019-04-18T22:05:37.9711684Z: 1d70e22e-efgh-458e-abcd-12345d34d34a - AdalLoggerBase.cs: A match was found in the MSAL cache ? False
Verbose 2019-04-18T22:05:37.9736674Z: 1d70e22e-efgh-458e-abcd-12345d34d34a - AdalLoggerBase.cs: Looking up cache for a token...
Information 2019-04-18T22:05:37.9756671Z: 1d70e22e-efgh-458e-abcd-12345d34d34a - AdalLoggerBase.cs: No matching token was found in the cache
Information 2019-04-18T22:05:37.9781677Z: 1d70e22e-efgh-458e-abcd-12345d34d34a - AdalLoggerBase.cs: Checking MSAL cache for user token cache
Information 2019-04-18T22:05:37.9811687Z: 1d70e22e-efgh-458e-abcd-12345d34d34a - AdalLoggerBase.cs: A match was found in the MSAL cache ? False
Verbose 2019-04-18T22:05:37.9836689Z: 1d70e22e-efgh-458e-abcd-12345d34d34a - AdalLoggerBase.cs: Either a token was not found or an exception was thrown.
Verbose 2019-04-18T22:05:37.9861725Z: 1d70e22e-efgh-458e-abcd-12345d34d34a - AdalLoggerBase.cs: Cannot invoke the broker directly, may require install ...

Verbose 2019-04-18T22:05:57.0596726Z: 1d70e22e-efgh-458e-abcd-12345d34d34a - AdalLoggerBase.cs: Check and AcquireToken using broker 
Verbose 2019-04-18T22:05:57.0616748Z: 1d70e22e-efgh-458e-abcd-12345d34d34a - AdalLoggerBase.cs: Broker invocation is NOT required
Verbose 2019-04-18T22:05:57.3342319Z: 1d70e22e-efgh-458e-abcd-12345d34d34a - AdalLoggerBase.cs: Storing token in the cache...
Verbose 2019-04-18T22:05:57.3387337Z: 1d70e22e-efgh-458e-abcd-12345d34d34a - AdalLoggerBase.cs: An item was stored in the cache
Information 2019-04-18T22:05:57.3707324Z: 00000000-0000-0000-0000-000000000000 - AdalLoggerBase.cs: Serializing token cache with 1 items.
Information 2019-04-18T22:05:57.4212318Z: 1d70e22e-efgh-458e-abcd-12345d34d34a - AdalLoggerBase.cs: === Token Acquisition finished successfully. An access token was returned: Expiration Time: 4/18/2019 11:05:57 PM +00:00Access Token Hash: XNXNoxHpErX23eV8gjS3g/abcagaaNJs1t4hK9ZF816pzk=
	 User id: 7cfeebd7-56af-1234-8eac-10a0ee703dfa

Information 2019-04-18T22:06:02.2806658Z: ab1493d2-21fe-1234-b18c-5d4f515be9ab - AdalLoggerBase.cs: ADAL PCL.UAP with assembly version '5.0.1.0', file version '5.0.1.0' and informational version '5.0.1-internal10701ssofix' is running...
Information 2019-04-18T22:06:02.2830572Z: ab1493d2-21fe-1234-b18c-5d4f515be9ab - AdalLoggerBase.cs: === Token Acquisition started:
	Authority: https://login.microsoftonline.com/Tenant/
	Resource: https://xyz.com/Technician-WS-DEV
	ClientId: 299a1a70-xxxx-4c56-xxx-85a344xxx955c
	CacheType: Microsoft.IdentityModel.Clients.ActiveDirectory.TokenCache (1 items)
	Authentication Target: User
	
Verbose 2019-04-18T22:06:02.2865546Z: ab1493d2-21fe-1234-b18c-5d4f515be9ab - AdalLoggerBase.cs: Loading from cache.
Information 2019-04-18T22:06:02.3115535Z: 00000000-0000-0000-0000-000000000000 - AdalLoggerBase.cs: Deserialized 1 items to token cache.
Verbose 2019-04-18T22:06:02.3145532Z: ab1493d2-21fe-1234-b18c-5d4f515be9ab - AdalLoggerBase.cs: Looking up cache for a token...
Information 2019-04-18T22:06:02.3170594Z: ab1493d2-21fe-1234-b18c-5d4f515be9ab - AdalLoggerBase.cs: An item matching the requested resource was found in the cache
Verbose 2019-04-18T22:06:02.3200551Z: ab1493d2-21fe-1234-b18c-5d4f515be9ab - AdalLoggerBase.cs: A matching entry was found in the cache
Information 2019-04-18T22:06:02.3230552Z: ab1493d2-21fe-1234-b18c-5d4f515be9ab - AdalLoggerBase.cs: 59.91659908 minutes left until token in cache expires
Information 2019-04-18T22:06:02.3250571Z: ab1493d2-21fe-1234-b18c-5d4f515be9ab - AdalLoggerBase.cs: A matching item (access token or refresh token or both) was found in the cache
Information 2019-04-18T22:06:02.3280584Z: ab1493d2-21fe-1234-b18c-5d4f515be9ab - AdalLoggerBase.cs: === Token Acquisition finished successfully. An access token was returned: Expiration Time: 4/18/2019 11:05:57 PM +00:00Access Token Hash: XNXNoxHpErX23eV8gjS3g/abcagaaNJs1t4hK9ZF816pzk=
	 User id: 7cfeebd7-56af-1234-8eac-10a0ee703dfa
Information 2019-04-18T22:06:02.4435535Z: 451ccba8-c167-4ec9-ab96-f443f635e8fe - AdalLoggerBase.cs: ADAL PCL.UAP with assembly version '5.0.1.0', file version '5.0.1.0' and informational version '5.0.1-internal10701ssofix' is running...
Information 2019-04-18T22:06:02.4460565Z: 451ccba8-c167-7890-ab96-f443f635e8fe - AdalLoggerBase.cs: === Token Acquisition started:
	Authority: https://login.microsoftonline.com/Tenant/
	Resource: https://xyz.com/Technician-WS-DEV
	ClientId: 299a1a70-xxxx-4c56-xxx-85a344xxx955c
	CacheType: Microsoft.IdentityModel.Clients.ActiveDirectory.TokenCache (1 items)
	Authentication Target: User
	
Verbose 2019-04-18T22:06:02.4505533Z: 451ccba8-c167-7890-ab96-f443f635e8fe - AdalLoggerBase.cs: Loading from cache.
Information 2019-04-18T22:06:02.4560546Z: 00000000-0000-0000-0000-000000000000 - AdalLoggerBase.cs: Deserialized 1 items to token cache.
Verbose 2019-04-18T22:06:02.4585529Z: 451ccba8-c167-7890-ab96-f443f635e8fe - AdalLoggerBase.cs: Looking up cache for a token...
Information 2019-04-18T22:06:02.4610532Z: 451ccba8-7890-4ec9-ab96-f443f635e8fe - AdalLoggerBase.cs: An item matching the requested resource was found in the cache
Verbose 2019-04-18T22:06:02.4635534Z: 451ccba8-7890-4ec9-ab96-f443f635e8fe - AdalLoggerBase.cs: A matching entry was found in the cache
Information 2019-04-18T22:06:02.4665535Z: 451ccba8-7890-4ec9-ab96-f443f635e8fe - AdalLoggerBase.cs: 59.9142074416667 minutes left until token in cache expires
Information 2019-04-18T22:06:02.4690559Z: 451ccba8-7890-4ec9-ab96-f443f635e8fe - AdalLoggerBase.cs: A matching item (access token or refresh token or both) was found in the cache
Information 2019-04-18T22:06:02.4715539Z: 451ccba8-7890-4ec9-ab96-f443f635e8fe - AdalLoggerBase.cs: === Token Acquisition finished successfully. An access token was returned: Expiration Time: 4/18/2019 11:05:57 PM +00:00Access Token Hash: XNXNoxHpErX23eV8gjS3g/abcagaaNJs1t4hK9ZF816pzk=
	 User id: 7cfeebd7-56af-1234-8eac-10a0ee703dfa
Information 2019-04-18T22:06:09.5723770Z: 00000000-0000-0000-0000-000000000000 - AdalLoggerBase.cs: Deserialized 1 items to token cache.
Information 2019-04-18T22:06:09.5748791Z: 00000000-0000-0000-0000-000000000000 - AdalLoggerBase.cs: Clearing Cache :- 1 items to be removed
Information 2019-04-18T22:06:09.5808827Z: 00000000-0000-0000-0000-000000000000 - AdalLoggerBase.cs: Successfully Cleared Cache
Information 2019-04-18T22:06:09.5848785Z: 00000000-0000-0000-0000-000000000000 - AdalLoggerBase.cs: Serializing token cache with 0 items.
Information 2019-04-18T22:06:11.5111650Z: c1b5bc28-xyz1-43c6-aa43-33301842d89a - AdalLoggerBase.cs: ADAL PCL.UAP with assembly version '5.0.1.0', file version '5.0.1.0' and informational version '5.0.1-internal10701ssofix' is running...
Information 2019-04-18T22:06:11.5142015Z: c1b5bc28-xyz1-43c6-aa43-33301842d89a - AdalLoggerBase.cs: === Token Acquisition started:
	Authority: https://login.microsoftonline.com/Tenant/
	Resource: https://xyz.com/Technician-WS-DEV
	ClientId: 299a1a70-xxxx-4c56-xxx-85a344xxx955c
	CacheType: Microsoft.IdentityModel.Clients.ActiveDirectory.TokenCache (0 items)
	Authentication Target: User
	
Verbose 2019-04-18T22:06:11.5167011Z: c1b5bc28-xyz1-43c6-aa43-33301842d89a - AdalLoggerBase.cs: Loading from cache.
Information 2019-04-18T22:06:11.5282034Z: 00000000-0000-0000-0000-000000000000 - AdalLoggerBase.cs: Deserialized 0 items to token cache.
Verbose 2019-04-18T22:06:11.5312051Z: c1b5bc28-xyz1-43c6-aa43-33301842d89a - AdalLoggerBase.cs: Looking up cache for a token...
Information 2019-04-18T22:06:11.5336991Z: c1b5bc28-xyz1-43c6-aa43-33301842d89a - AdalLoggerBase.cs: No matching token was found in the cache
Information 2019-04-18T22:06:11.5372074Z: c1b5bc28-xyz1-43c6-aa43-33301842d89a - AdalLoggerBase.cs: Checking MSAL cache for user token cache
Information 2019-04-18T22:06:11.5407002Z: c1b5bc28-xyz1-43c6-aa43-33301842d89a - AdalLoggerBase.cs: A match was found in the MSAL cache ? False
Verbose 2019-04-18T22:06:11.5432001Z: c1b5bc28-xyz1-43c6-aa43-33301842d89a - AdalLoggerBase.cs: Looking up cache for a token...
Information 2019-04-18T22:06:11.5462012Z: c1b5bc28-xyz1-43c6-aa43-33301842d89a - AdalLoggerBase.cs: No matching token was found in the cache
Information 2019-04-18T22:06:11.5489568Z: c1b5bc28-xyz1-43c6-aa43-33301842d89a - AdalLoggerBase.cs: Checking MSAL cache for user token cache
Information 2019-04-18T22:06:11.5524558Z: c1b5bc28-xyz1-43c6-aa43-33301842d89a - AdalLoggerBase.cs: A match was found in the MSAL cache ? False
Verbose 2019-04-18T22:06:11.5554621Z: c1b5bc28-xyz1-43c6-aa43-33301842d89a - AdalLoggerBase.cs: Looking up cache for a token...
Information 2019-04-18T22:06:11.5579604Z: c1b5bc28-xyz1-43c6-aa43-33301842d89a - AdalLoggerBase.cs: No matching token was found in the cache
Information 2019-04-18T22:06:11.5604575Z: c1b5bc28-xyz1-43c6-aa43-33301842d89a - AdalLoggerBase.cs: Checking MSAL cache for user token cache
Information 2019-04-18T22:06:11.5639578Z: c1b5bc28-xyz1-43c6-aa43-33301842d89a - AdalLoggerBase.cs: A match was found in the MSAL cache ? False
Verbose 2019-04-18T22:06:11.5664571Z: c1b5bc28-xyz1-43c6-aa43-33301842d89a - AdalLoggerBase.cs: Looking up cache for a token...
Information 2019-04-18T22:06:11.5694643Z: c1b5bc28-xyz1-43c6-aa43-33301842d89a - AdalLoggerBase.cs: No matching token was found in the cache
Information 2019-04-18T22:06:11.5719568Z: c1b5bc28-xyz1-43c6-aa43-33301842d89a - AdalLoggerBase.cs: Checking MSAL cache for user token cache
Information 2019-04-18T22:06:11.5754577Z: c1b5bc28-xyz1-43c6-aa43-33301842d89a - AdalLoggerBase.cs: A match was found in the MSAL cache ? False
Verbose 2019-04-18T22:06:11.5779585Z: c1b5bc28-xyz1-43c6-aa43-33301842d89a - AdalLoggerBase.cs: Looking up cache for a token...
Information 2019-04-18T22:06:11.5804581Z: c1b5bc28-xyz1-43c6-aa43-33301842d89a - AdalLoggerBase.cs: No matching token was found in the cache
Information 2019-04-18T22:06:11.5829562Z: c1b5bc28-xyz1-43c6-aa43-33301842d89a - AdalLoggerBase.cs: Checking MSAL cache for user token cache
Information 2019-04-18T22:06:11.5859625Z: c1b5bc28-xyz1-43c6-aa43-33301842d89a - AdalLoggerBase.cs: A match was found in the MSAL cache ? False
Verbose 2019-04-18T22:06:11.5884580Z: c1b5bc28-xyz1-43c6-aa43-33301842d89a - AdalLoggerBase.cs: Looking up cache for a token...
Information 2019-04-18T22:06:11.5909604Z: c1b5bc28-xyz1-43c6-aa43-33301842d89a - AdalLoggerBase.cs: No matching token was found in the cache
Information 2019-04-18T22:06:11.5939605Z: c1b5bc28-xyz1-43c6-aa43-33301842d89a - AdalLoggerBase.cs: Checking MSAL cache for user token cache
Information 2019-04-18T22:06:11.5969631Z: c1b5bc28-xyz1-43c6-aa43-33301842d89a - AdalLoggerBase.cs: A match was found in the MSAL cache ? False
Verbose 2019-04-18T22:06:11.5994593Z: c1b5bc28-xyz1-43c6-aa43-33301842d89a - AdalLoggerBase.cs: Either a token was not found or an exception was thrown.
Verbose 2019-04-18T22:06:11.6019604Z: c1b5bc28-xyz1-43c6-aa43-33301842d89a - AdalLoggerBase.cs: Cannot invoke the broker directly, may require install ...

```

**Comment by jennyf19** [comment]
@Arslan007 Thank you!

@bgavrilMS I asked Arslan to open a new issue here related only to the UWP simulator, for tracking purposes. The [fix we put out works on devices](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/issues/1580), and that unblocks his team so they can release, but we should delve into this simulator issue at some point.

cc: @jmprieur

**Comment by bgavrilMS** [comment]
Thanks, I updated the wiki.
